util/test/demos/gl/gl_backdoor_context.cpp

```diff
@@ -0,0 +1,178 @@
+layout(location = 2) in vec2 UV;
+
+out v2f vertOut;
+
+void main()
+{
+	vertOut.pos = vec4(Position.xyz, 1);
+	gl_Position = vertOut.pos;
+	vertOut.col = Color;
+	vertOut.uv = vec4(UV.xy, 0, 1);
+}
+
+)EOSHADER";
+
+  std::string pixel = R"EOSHADER(
+
+in v2f vertIn;
+
+layout(location = 0, index = 0) out vec4 Color;
+
+layout(binding = 0) uniform sampler2D tex;
+
+void main()
+{
+	Color = vertIn.col * texture(tex, vertIn.uv.xy);
+}
+
+)EOSHADER";
+
+  int main()
+  {
+    // initialise, create window, create context, etc
+    if(!Init())
+      return 3;
+
+	GLuint vao, vb, program;
+
+    void *contextA = mainContext;
+    void *contextB = MakeContext(mainWindow, NULL);
+
+    GLuint fboB, texB;
+    ActivateContext(mainWindow, contextB);
+    {
+      vao = MakeVAO();
+      glBindVertexArray(vao);
+
+      vb = MakeBuffer();
+      glBindBuffer(GL_ARRAY_BUFFER, vb);
+      glBufferStorage(GL_ARRAY_BUFFER, sizeof(DefaultTri), DefaultTri, 0);
+
+      glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, sizeof(DefaultA2V), (void *)(0));
+      glVertexAttribPointer(1, 4, GL_FLOAT, GL_FALSE, sizeof(DefaultA2V), (void *)(sizeof(Vec3f)));
+      glVertexAttribPointer(2, 2, GL_FLOAT, GL_FALSE, sizeof(DefaultA2V),
+                            (void *)(sizeof(Vec3f) + sizeof(Vec4f)));
+
+      glEnableVertexAttribArray(0);
+      glEnableVertexAttribArray(1);
+      glEnableVertexAttribArray(2);
+
+      program = MakeProgram(common + vertex, common + pixel);
+
+      glGenFramebuffers(1, &fboB);
+      glBindFramebuffer(GL_FRAMEBUFFER, fboB);
+
+      glGenTextures(1, &texB);
+      glBindTexture(GL_TEXTURE_2D, texB);
+
+      glTextureStorage2D(texB, 1, GL_RGBA8, screenWidth, screenHeight);
+      glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, texB, 0);
+
+	  glBindFramebuffer(GL_FRAMEBUFFER, fboB);
+
+      float col[] = {1.0f, 0.5f, 0.25f, 1.0f};
+      glClearBufferfv(GL_COLOR, 0, col);
+    }
+
+	// wait 5ms to make sure texB becomes a persistent resource
+    clock_t start_t = clock();
+    while((double)(clock() - start_t) / CLOCKS_PER_SEC < 0.005)
+      Sleep(0);
+
+    while(Running())
+    {
+      ActivateContext(mainWindow, contextA);
+      {
+        glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+        float col[] = {1.0f, 0.0f, 0.0f, 1.0f};
+        glClearBufferfv(GL_COLOR, 0, col);
+      }
+
+	  ActivateContext(mainWindow, contextB);
+      {
+        glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+        glBindTextureUnit(0, texB);
+
+        glBindVertexArray(vao);
+
+        glUseProgram(program);
+
+        glViewport(0, 0, GLsizei(screenWidth), GLsizei(screenHeight));
+
+        glDrawArrays(GL_TRIANGLES, 0, 3);
+      }
+
+      ActivateContext(mainWindow, contextA);
+      {
+        glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+        float col[] = {0.0f, 1.0f, 0.0f, 1.0f};
+        glClearBufferfv(GL_COLOR, 0, col);
+      }
+
+      Present();
+    }
+
+    glDeleteTextures(1, &texB);
```

Comment:
You need to activate `contextB` before deleting B's resources. This will also include the program/vao/buffer as well.

You'll also need to add a fix into `OpenGLGraphicsTest::Shutdown` to activate the main context before shutting down and destroying any resources (since this will leave `contextB` active).